### PR TITLE
Vagrantfile: enable virtio & KVM paravirt for L3

### PR DESCRIPTION
**Pull request by unclejack**
This modifies the Vagrantfile to change the NIC type to virtio for the quagga VMs. It also sets the KVM paravirt for these VMs.

A function is added to avoid duplicating those lines for the netplugin nodes and for the two quagga VMs.

The quagga VMs NICs used to be `Intel Corporation 82540EM Gigabit Ethernet Controller` before this change. They all show up as virtio with this change.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -121,6 +121,17 @@ echo "export https_proxy='$2'" >> /etc/profile.d/envvar.sh
 source /etc/profile.d/envvar.sh
 SCRIPT
 
+def customize(v, id)
+  # make all nics 'virtio' to take benefit of builtin vlan tag
+  # support, which otherwise needs to be enabled in Intel drivers,
+  # which are used by default by virtualbox
+  # changes the settings for the first 5 NICs, regardless of presence
+  1.upto(5) do |n|
+    v.customize ['modifyvm', :id, '--nictype' + n.to_s , 'virtio']
```
Comment (dseevr) [inline Vagrantfile]: If you use a doublequoted string + interpolation, #to_s will automatically be called on the interpolated object:

```rb
"--nictype#{n}"
```

**Comment by unclejack** [comment]
@dseevr I've made the change. It works. PTAL

**Comment by dseevr** [comment]
LGTM 👍

**Comment by unclejack** [comment]
@jojimt Please take a look at this when you have the time.

**Pull request merged**